geckomat/utilities/ecFSEOF/run_ecFSEOF.m

```diff
@@ -0,0 +1,69 @@
+function results = run_ecFSEOF(model,rxnTarget,cSource,alphaLims,Nsteps,file1,file2)
+% 
+% run_ecFSEOF
+%
+% 	Function that runs Flux-scanning with Enforced Objective Function
+% 	for a specified production target.
+%   
+%		model     (struct) ecModel with total protein pool constraint
+%		rxnTarget (string) Rxn ID for the production target reaction, 
+% 				  a exchange reaction is recommended.
+%		cSource	  (string) Rxn name for the main carbon source uptake 
+%			      reaction
+% 		alphaLims (vector) Minimum and maximum biomass yield [gDw/mmol Csource] 
+%				  for enforced objective limits
+%       Nsteps    (integer) Number of steps for suboptimal objective 
+%                 in FSEOF
+%		file1 	  (string) opt, file name for an output .txt tab-separated 
+% 				  file for the results at the genes level
+%		file2 	  (string) opt, file name for an output .txt tab-separated 
+% 				  file for the results at the reactions level		
+%
+% Usage: run_ecFSEOF(ecModel,rxnTarget,cSource,alphaLims,Nsteps)
+%
+% Last modified.  Ivan Domenzain 2019-11-13
+%
+
+if nargin < 7
+	file2 = [];
+	if nargin < 6
+		file1 = [];
+	end
+end
+%Define alpha vector for suboptimal enforced objective values
+alphaV  = alphaLims(1):((alphaLims(2)-alphaLims(1))/(Nsteps-1)):alphaLims(2);
+%Standardize grRules and rxnGeneMat in model
+[grRules,rxnGeneMat] = standardizeGrRules(model,true);
+model.grRules        = grRules;
+model.rxnGeneMat     = rxnGeneMat;
+%run FSEOF analysis
+results = ecFlux_scanning(model,rxnTarget,cSource,alphaV,1E-4,true);
+%Create gene table:
+results.geneTable      = cell(length(results.genes),3);
+results.geneTable(:,1) = results.genes;
+results.geneTable(:,2) = results.geneNames;
+results.geneTable(:,3) = num2cell(results.k_genes);
+%Create rxns table (exclude enzyme usage reactions):
```

Comment:
Just a thought here: Why exclude enzyme usage reactions from the analysis? Considering that those reactions translate 1-1 to genes, could it be an alternative way of getting gene candidates? I ask because in other FSEOF analysis I've conducted on ecModels (for instance [this](https://github.com/BenjaSanchez/notebooks/blob/master/post-ec-model-biotin/notebook.ipynb) one), I have **_only_** looked at the enzyme usage reactions.

Also, I noticed that in the case study I ran for `ecYeast` when testing this branch, there were more enzyme usage reactions (562-304=258) than genes (126), have you seen this as well in your tests?

Reply:
> Just a thought here: Why exclude enzyme usage reactions from the analysis? Considering that those reactions translate 1-1 to genes, could it be an alternative way of getting gene candidates? I ask because in other FSEOF analysis I've conducted on ecModels (for instance [this](https://github.com/BenjaSanchez/notebooks/blob/master/post-ec-model-biotin/notebook.ipynb) one), I have **_only_** looked at the enzyme usage reactions.
> 
> Also, I noticed that in the case study I ran for `ecYeast` when testing this branch, there were more enzyme usage reactions (562-304=258) than genes (126), have you seen this as well in your tests?

Very interesting comment @BenjaSanchez! I've addressed this (partially) in fbad8c6. Short summary:

It is strange that the algorithm returns more `draw_prot_XXXX` reactions than genes in the final `geneTable`. I checked the code and remembered that there is a step that removes all of those gene targets considered as "unchanged" (gene targets with a  `mean(K_score) < K_score < 1`), which makes sense in order to just report the most relevant results. Therefore the mismatch in the numbers was caused by those remaining `draw_prot_XXXX` rxns in the `rxnTable` associated to genes that are removed in the aforementioned step. As a fix, commit fbad8c6 updates the rxn-related fields, making sure that any rxns related to the removed "unchanged" genes are also removed from the results.

Reply:
> Just a thought here: Why exclude enzyme usage reactions from the analysis? Considering that those reactions translate 1-1 to genes, could it be an alternative way of getting gene candidates? I ask because in other FSEOF analysis I've conducted on ecModels (for instance [this](https://github.com/BenjaSanchez/notebooks/blob/master/post-ec-model-biotin/notebook.ipynb) one), I have **_only_** looked at the enzyme usage reactions.
> 
> Also, I noticed that in the case study I ran for `ecYeast` when testing this branch, there were more enzyme usage reactions (562-304=258) than genes (126), have you seen this as well in your tests?

As a follow up to the initial question (@BenjaSanchez): The enzyme usage reactions are not excluded from the analysis, they are just removed from the final list of rxns results (trying to just show metabolic reactions for simplicity). However, these reactions are accounted for in previous steps (inside of the `ecFlux_scanning` function), therefore the genes associated to them should be included in the `genesTable` results. The intention is to keep results separated by rxns and genes, both having different fields however their connection is described in the column `grRules` in the `rxnsTable`.

Even though the`draw_prot_XXXX` reactions provide a 1-1 relation to genes, there is risk of ignoring potential gene targets that are not directly associated to any enzyme in the ecModel structure. This problem is not so acute for `ecYeastGEM` in which the enzyme coverage is high, but it might be relevant for industrially relevant species whose corresponding ecModels present a lower enzyme coverage (e.g. *Y. lipolytica* and *K. marxianus*).

Reply:
Roger that! all good then, thanks for looking into it.